docs/src/concepts/kernels.md

```diff
@@ -0,0 +1,93 @@
+
+Followingly, we show how to write and launch kernels on various backends. We also explain the concept of a `Launcher` in [Chmy.jl](https://github.com/PTsolvers/Chmy.jl), that complements the default kernel launching, allowing us to perform more fine-grained operations on different `Field` objects.
+
+## Writing Kernels
+
+This section highlights some important features of [KernelAbstactions.jl](https://github.com/JuliaGPU/KernelAbstractions.jl) that are essential for understanding the high-level abstraction of the kernel concept that is used throughout our package. As it barely serves for illustrative purposes, for more specific examples, please refer to their [documentation](https://juliagpu.github.io/KernelAbstractions.jl/stable/).
+
+
+```julia
+using KernelAbstactions
+
+# Define a kernel that performs element-wise operations on A
+@kernel function mul2(A)
+    I = @index(Global, Cartesian)
+    A[I] *= 2
+end
+```
+
+With the kernel `mul2` as defined above, we can launch the kernel to perform the element-wise operations on host.
+
+```julia
+# Define array and work group size
+workgroup_size = 64
+A              = ones(1024, 1024)
+backend        = get_backend(A) # CPU
+
+# Launch kernel and explicitly synchronize
+mul2(backend, workgroup_size)(A, ndrange=size(A))
+synchronize(backend)
+
+# Result assertion
+@assert(all(A .== 2.0) == true)
+```
+
+To launch the kernel on GPU devices, one could simply define `A` as `CuArray`, `ROCArray` or `oneArray` as detailed in the section ["launching kernel on the backend"](https://juliagpu.github.io/KernelAbstractions.jl/stable/quickstart/#Launching-kernel-on-the-backend). More fine-grained memory access is available using the `@index` macro as described [here](https://juliagpu.github.io/KernelAbstractions.jl/stable/api/#KernelAbstractions.@index).
+
+
+!!! info "Kernel Synchronization"
+    In previous section about [task-based parallelism](workers.md), we have mentioned and showed the advantages of asynchronous execution. When it comes to using both CPU and GPU, it appears natural that by default kernel launches are asynchronous. Therefore, in [KernelAbstactions.jl](https://github.com/JuliaGPU/KernelAbstractions.jl), one also needs to explicitly use `synchronize(backend)` function to wait on a series of kernel launches.
+
+
+## Kernel Launcher
+
+In [Chmy.jl](https://github.com/PTsolvers/Chmy.jl), the `KernelLaunch` module is designed to provide handy utilities for performing different grid operations on selected data entries of `Field`s that are involved at each kernel launch, in which the grid geometry underneath is also taken into account.
+
+Followingly, we define a kernel launcher associated with an `UniformGrid` object, supporting CUDA backend.
+
+```julia
+# Define backend and geometry
+arch   = Arch(CUDABackend())
+grid   = UniformGrid(arch; origin=(-1, -1), extent=(2, 2), dims=(126, 126))
+
+# Define launcher
+launch = Launcher(arch, grid; outer_width=(16, 8))
```

Comment:
I would maybe not use the `outer_width` here, unless one first also describes its behaviour, and the concept of async execution.